driver/src/main/java/org/neo4j/driver/internal/cluster/RoundRobinAddressSet.java

```diff
@@ -0,0 +1,129 @@
+/**
+ * Copyright (c) 2002-2016 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.neo4j.driver.internal.cluster;
+
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.neo4j.driver.internal.net.BoltServerAddress;
+
+class RoundRobinAddressSet extends AtomicInteger
+{
+    private static final BoltServerAddress[] NONE = {};
+    private volatile BoltServerAddress[] addresses = NONE;
+
+    public int size()
+    {
+        return addresses.length;
+    }
+
+    public BoltServerAddress next()
```

Comment:
I kind of liked that it used to have a name that wasn't `next` - although `hop` maybe wasn't ideal - the behaviour of the method is quite different from `Iterator.next` so having it named differently makes that more obvious.